.github/workflows/full-build.yml

```diff
@@ -145,63 +189,21 @@ jobs:
-      - name: Import codesign certificate
-        uses: apple-actions/import-codesign-certs@v1.0.4
-        with:
-          p12-file-base64: ${{ secrets.BASE64_ENCODED_P12 }}
-          p12-password: ${{ secrets.CERTPASSWORD_P12 }}
-      - name: Create Signed macOS Application
-        run: poetry run poe package
-      - name: Notarize app
-        uses: devbotsxyz/xcode-notarize@v1.0.0
-        with:
-          product-path: "_packaging/dist/Gaphor.app"
-          appstore-connect-username: ${{ secrets.AC_USERNAME }}
-          appstore-connect-password: ${{ secrets.AC_PASSWORD }}
-      - name: Staple app
-        uses: devbotsxyz/xcode-staple@v1.0.0
-        with:
-          product-path: "_packaging/dist/Gaphor.app"
-      - name: Create dmg
-        run: >
-          cd _packaging
-
-          create-dmg --volname "Gaphor ${{ steps.setup_and_test.outputs.version }}"
-          --background "macos/background.png"
-          --window-pos 200 120 --window-size 700 400 --icon-size 100
-          --icon "Gaphor.app" 200 240 --hide-extension "Gaphor.app"
-          --app-drop-link 500 240 "dist/Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg"
-          "dist/Gaphor.app"
-      - name: Notarize dmg
-        uses: devbotsxyz/xcode-notarize@v1
-        with:
-          product-path: "_packaging/dist/Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg"
-          appstore-connect-username: ${{ secrets.AC_USERNAME }}
-          appstore-connect-password: ${{ secrets.AC_PASSWORD }}
-          primary-bundle-id: org.gaphor.gaphor
-      - name: Staple .dmg
-        uses: devbotsxyz/xcode-staple@v1.0.0
-        with:
-          product-path: "_packaging/dist/Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg"
-      - name: Upload Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg
-        uses: actions/upload-artifact@v3
-        with:
-          name: Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg
-          path: _packaging/dist/Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg
-      - name: Upload Assets (release only)
-        uses: AButler/upload-release-assets@v2.0
-        if: github.event_name == 'release'
+      - name: Create macOS Application
+        uses: ./.github/actions/macos_dmg
         with:
-          files: "_packaging/dist/Gaphor-${{ steps.setup_and_test.outputs.version }}.dmg"
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          mainline_build: ${{ env.mainline_build }}
+          version: ${{ steps.setup_and_test.outputs.version }}
+          base64_encoded_p12: ${{ secrets.BASE64_ENCODED_P12 }}
+          certpassword_p12: ${{ secrets.CERTPASSWORD_P12 }}
+          ac_username:  ${{ secrets.AC_USERNAME }}
+          ac_password: ${{ secrets.AC_PASSWORD }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
 
   check-macos-dmg:
     needs: macos
     runs-on: macos-latest
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/packaging/')
```

Comment:
Should we just make this `if: ${{ mainline_build == 'true' }}`?

Reply:
That doesn't work: mainline_build is an env var and therefore can't be used in those job level if statements. Sadly.